src/Bridge/RamseyUuid/Identifier/Normalizer/UuidNormalizer.php

```diff
@@ -28,20 +28,22 @@ final class UuidNormalizer implements DenormalizerInterface
 {
     /**
      * {@inheritdoc}
+     *
+     * @throws NotNormalizableValueException Occurs when the identifier format is invalid
      */
     public function denormalize($data, $class, $format = null, array $context = [])
     {
         try {
             return Uuid::fromString($data);
         } catch (InvalidUuidStringException $e) {
-            throw new InvalidIdentifierException($e->getMessage(), $e->getCode(), $e);
+            throw new NotNormalizableValueException('Not found, because of an invalid identifier format', 404, $e);
```

Comment:
This exception doesn't take an http status code as second argument. I'd still keep the `$e->getMessage()` error but that's my opinion.

Reply:
Hey @soyuka, I initially added the http status code instead of `$e->getMessage()` since the Behat test (`features/main/uuid.feature`) demands that it throws an 404 error (which would be the correct code) but when I run the test suite it returns 400 instead and I don't know why. Any ideas?

Reply:
Because you changed the exception... InvalidIdentifierException is the correct one imo.  I'm not sure we should change that...

See ReadListener (I think) where InvalidIdentifierException is switched to a 404.

Reply:
Ah! Thank you! @teohhanhui mentioned this exception as "it should be the correct one". TBH sounds `InvalidIdentifierException` like the correct exception here when you take the new exception message into account.

Reply:
Yes, `NotNormalizableValueException` is the correct one. We should not throw our own exception class here, as this is a (de)normalizer.

See https://github.com/symfony/symfony/blob/v4.3.4/src/Symfony/Component/Serializer/Normalizer/DenormalizerInterface.php#L45

Reply:
Then we need to catch it [here](https://github.com/api-platform/core/blob/8d625d00404958d90dd6f94f84ccec26be30dd72/src/EventListener/ReadListener.php#L112) and throw a notfound but I'm afraid that it'll interfer with what should be a 400.

At first when I initiated this class I didn't wanted to use the symfony interface because they're not used in the serializer but in the identifier normalization phase.

Reply:
You have a point. Anyway, this PR is not the right fix for #3045.

IMO, we don't need to change anything here. Just change the error message in the `ReadListener` so that it's not misleading:

https://github.com/api-platform/core/blob/81f819b3befe1f0bb726e1901f4c8c6e500bb60e/src/EventListener/ReadListener.php#L112

`Invalid identifier value or configuration.`

Reply:
exactly